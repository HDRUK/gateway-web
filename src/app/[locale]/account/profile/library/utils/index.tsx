import { IconButton } from "@mui/material";
import { createColumnHelper } from "@tanstack/react-table";
import { LibraryListItem, SelectedLibrary } from "@/interfaces/Library";
import Link from "@/components/Link";
import StyledCheckbox from "@/components/StyledCheckbox";
import TooltipIcon from "@/components/TooltipIcon";
import { CheckIcon, DeleteForeverIcon } from "@/consts/icons";
import { RouteName } from "@/consts/routeName";

const columnHelper = createColumnHelper<LibraryListItem>();

const getColumns = ({
    handleSelect,
    handleRemove,
    selected,
    translations,
}: {
    handleSelect: (data: SelectedLibrary) => void;
    handleRemove: (id: string | number) => void;
    selected: SelectedLibrary;
    translations: { [id: string]: string };
}) => [
    columnHelper.display({
        id: "actions",
        meta: { isPinned: true },
        cell: ({ row }) => {
            const {
                id,
                datasetId,
                name,
                dataCustodianId,
                dataCustodian,
                darEnabled,
                darTemplatePublished,
                cohortEnabled,
            } = row.original;
            return (
                <div style={{ textAlign: "center" }}>
                    <StyledCheckbox
                        checked={selected[id]?.selected}
                        onChange={(_e, value) =>
                            handleSelect({
                                [id]: {
                                    selected: value,
                                    datasetId,
                                    name,
                                    teamId: dataCustodianId,
                                    teamName: dataCustodian,
                                    darEnabled,
                                    darTemplatePublished,
                                    cohortEnabled,
                                },
                            })
                        }
                        size="large"
                        sx={{ p: 0 }}
                        iconSx={{ mr: 0 }}
                    />
                </div>
            );
        },
        header: () => null,
        size: 43,
    }),

    columnHelper.display({
        id: "name",
        cell: ({ row: { original } }) => {
            const { datasetId, name } = original;
            const linkHref = `/${RouteName.DATASET_ITEM}/${datasetId}`;

            return (
                <Link href={linkHref}>
                    <div style={{ textAlign: "center" }}>{name}</div>
                </Link>
            );
        },
        header: () => <span>{translations.name}</span>,
        size: 150,
    }),

    columnHelper.display({
        id: "darEnabled",
        cell: ({
            row: {
                original: { darEnabled },
            },
        }) => (
            <div style={{ textAlign: "center" }}>
                {darEnabled ? <CheckIcon color="primary" /> : "-"}
            </div>
        ),
        header: () => (
            <TooltipIcon
                buttonSx={{ p: 0 }}
                size="small"
                label={translations.darEnabled}
                content={translations.darEnabled}
            />
        ),
        size: 100,
    }),

    columnHelper.display({
        id: "cohortEnabled",
        cell: ({
            row: {
                original: { cohortEnabled },
            },
        }) => (
            <div style={{ textAlign: "center" }}>
                {cohortEnabled ? <CheckIcon color="primary" /> : "-"}
            </div>
        ),
        header: () => <span>{translations.cohortEnabled}</span>,
        size: 100,
    }),

    columnHelper.display({
        id: "dataCustodian",
        cell: ({
            row: {
                original: { dataCustodian },
            },
        }) => <div style={{ textAlign: "center" }}>{dataCustodian}</div>,
        header: () => <span>{translations.dataCustodian}</span>,
        size: 100,
    }),

    columnHelper.display({
        id: "entityType",
        cell: ({
            row: {
                original: { entityType },
            },
        }) => <div style={{ textAlign: "center" }}>{entityType}</div>,
        header: () => <span>{translations.entityType}</span>,
        size: 100,
    }),

    columnHelper.display({
        id: "delete",
        cell: ({ row: { original } }) => {
            return (
                <div style={{ textAlign: "center" }}>
                    <IconButton onClick={() => handleRemove(original.id)}>
                        <DeleteForeverIcon color="primary" />
                    </IconButton>
                </div>
            );
        },
        header: () => null,
        size: 43,
    }),
];

export { getColumns };
