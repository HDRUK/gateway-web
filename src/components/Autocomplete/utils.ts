import { ValueType } from "./Autocomplete";

const getChipLabel = (
    options: { value: string | number; label: string }[],
    value: ValueType
) => {
    if (typeof value === "string") {
        return value;
    }

    return options.find(option => option.value === value)?.label;
};

export { getChipLabel };
