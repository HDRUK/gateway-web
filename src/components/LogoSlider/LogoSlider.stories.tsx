import type { Meta, StoryObj } from "@storybook/react";
import LogoSlider from "./LogoSlider";

const meta: Meta<typeof LogoSlider> = {
    component: LogoSlider,
    tags: ["autodocs"],
};

export default meta;

type Story = StoryObj<typeof LogoSlider>;

export const Default: Story = {
    args: {
        images: [
            {
                alt: "Microsoft",
                imageSrc: "/images/logos/microsoft-logo.png",
                websiteUrl: "http://www.google.com",
            },
            {
                alt: "LinkedIn",
                imageSrc: "/images/logos/linkedIn-logo.png",
                websiteUrl: "http://www.google.com",
            },
            {
                alt: "Google",
                imageSrc: "/images/logos/google-logo.png",
                websiteUrl: "http://www.google.com",
            },
        ],
    },
};
