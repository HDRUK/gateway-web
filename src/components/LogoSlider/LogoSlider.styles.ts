import { css, keyframes } from "@emotion/react";

const infiniteAnimation = keyframes`
  0% {
    transform: translate3d(0, 0, 0);
  }

  100% {
    transform: translate3d(-108rem, 0, 0);
  }

`;

export const ticker = css({
    display: "flex",
    "&:hover": { animationPlayState: "paused" },
    animation: `${infiniteAnimation} 18s linear infinite`,
});
