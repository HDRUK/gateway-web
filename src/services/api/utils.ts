import { NotificationOptions } from "@/interfaces/Api";
import { AxiosResponse } from "axios";
import { Error } from "@/interfaces/Error";
import notificationService from "../notification";

interface ErrorNotificationProps {
    props: NotificationOptions;
    errorResponse: AxiosResponse<Error>;
    method: "delete" | "post" | "put" | "get";
}

const errorNotification = ({
    errorResponse,
    method,
    props,
}: ErrorNotificationProps) => {
    const { t, i18n, ...notificationProps } = props;
    const { data, status } = errorResponse || {};

    const fallbackTitle = i18n.exists(`api:common.error.status.${status}`)
        ? t(`api:common.error.status.${status}`)
        : "There has been an error";

    const title = data?.title || fallbackTitle;
    const message =
        data?.message ||
        t(`api:common.error.${method}.message`, {
            item: props.itemName || "Item",
        });

    notificationService.apiError(message, {
        title,
        message,
        errors: data?.errors,
        ...notificationProps,
    });
};

interface SuccessNotificationProps {
    props: NotificationOptions;
    method: "delete" | "post" | "put";
}

const successNotification = ({ props, method }: SuccessNotificationProps) => {
    const { t, i18n, ...notificationProps } = props;
    const customMessage = `api:${props.localeKey}.success.${method}.message`;
    const shouldOverideMessage = i18n.exists(customMessage);

    const message = shouldOverideMessage
        ? t(customMessage)
        : t(`api:common.success.${method}.message`, {
              item: props.itemName || "Item",
          });

    notificationService.success(message, { ...notificationProps });
};

export { errorNotification, successNotification };
