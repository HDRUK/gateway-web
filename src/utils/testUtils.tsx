import React, { ReactNode } from "react";
import {
    render,
    renderHook,
    RenderHookOptions,
    RenderOptions,
    RenderResult,
    RenderHookResult,
} from "@testing-library/react";
import { SWRConfig } from "swr";
import { CacheProvider, ThemeProvider } from "@emotion/react";
import ActionBarProvider from "@/providers/ActionBarProvider";
import DialogProvider from "@/providers/DialogProvider";
import theme from "@/config/theme";
import createEmotionCache from "@/config/createEmotionCache";

const clientSideEmotionCache = createEmotionCache();

const Wrapper = ({ children }: { children: ReactNode }) => {
    return (
        <SWRConfig
            value={{
                provider: () => new Map(),
            }}>
            <CacheProvider value={clientSideEmotionCache}>
                <ThemeProvider theme={theme}>
                    <ActionBarProvider>
                        <DialogProvider>{children}</DialogProvider>
                    </ActionBarProvider>
                </ThemeProvider>
            </CacheProvider>
        </SWRConfig>
    );
};

interface OptionProps
    extends RenderOptions<
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement,
        HTMLElement
    > {
    wrapperProps: Record<string, unknown>;
}

const customRender = (
    ui: React.ReactElement<
        unknown,
        string | React.JSXElementConstructor<unknown>
    >,
    options?: OptionProps
): RenderResult => {
    const { wrapperProps, ...rest } = options || {};

    return render(ui, {
        wrapper: props => <Wrapper {...props} {...wrapperProps} />,
        ...rest,
    });
};

interface OptionHookProps
    extends RenderHookOptions<
        unknown,
        typeof import("@testing-library/dom/types/queries"),
        HTMLElement,
        HTMLElement
    > {
    wrapperProps: Record<string, unknown>;
}

const customRenderHook = (
    ui: (initialProps: unknown) => unknown,
    options?: OptionHookProps
): RenderHookResult<unknown, unknown> => {
    const { wrapperProps, ...rest } = options || {};
    return renderHook(ui, {
        wrapper: props => <Wrapper {...props} {...wrapperProps} />,
        ...rest,
    });
};

export * from "@testing-library/react";

export { customRenderHook as renderHook };
export { customRender as render };
