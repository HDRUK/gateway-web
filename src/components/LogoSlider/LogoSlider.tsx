/** @jsxImportSource @emotion/react */
import Image from "next/image";
import Box from "@/components/Box";
import * as styles from "./LogoSlider.styles";

interface LogoSliderProps {
    logos: { alt: string; imageSrc: string; websiteUrl: string }[];
}

const LogoSlider = ({ logos }: LogoSliderProps) => {
    return (
        <Box sx={{ background: "white", overflow: "hidden" }}>
            <div css={styles.ticker}>
                {logos.map(logo => (
                    <a
                        style={{
                            marginLeft: 10,
                            marginRight: 10,
                            width: 100,
                            flex: "none",
                            alignSelf: "flex-start",
                        }}
                        href={logo.websiteUrl}
                        target="_blank"
                        rel="noreferrer">
                        <Image
                            width={100}
                            height={50}
                            src={logo.imageSrc}
                            alt={logo.alt}
                        />
                    </a>
                ))}
            </div>
        </Box>
    );
};

export default LogoSlider;
