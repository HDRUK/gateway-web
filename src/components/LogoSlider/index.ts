import LogoSlider from "./LogoSlider";

export default LogoSlider;
