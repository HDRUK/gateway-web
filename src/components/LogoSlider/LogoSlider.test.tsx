import React from "react";
import LogoSlider from "@/components/LogoSlider";
import { render } from "@/utils/testUtils";

describe("LogoSlider", () => {
    it("should render component", async () => {
        const wrapper = render(
            <LogoSlider
                logos={[
                    {
                        alt: "Microsoft",
                        imageSrc: "/images/logos/microsoft-logo.png",
                        websiteUrl: "http://www.google.com",
                    },
                    {
                        alt: "LinkedIn",
                        imageSrc: "/images/logos/linkedIn-logo.png",
                        websiteUrl: "http://www.google.com",
                    },
                    {
                        alt: "Google",
                        imageSrc: "/images/logos/google-logo.png",
                        websiteUrl: "http://www.google.com",
                    },
                ]}
            />
        );

        expect(wrapper.container).toMatchSnapshot();
    });
});
