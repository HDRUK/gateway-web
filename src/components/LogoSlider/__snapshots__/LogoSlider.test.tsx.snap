// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoSlider should render component 1`] = `
<div>
  <div
    class="MuiBox-root mui-style-c5qhx0"
  >
    <div
      class="mui-style-vtvusl-LogoSlider"
    >
      <a
        href="http://www.google.com"
        rel="noreferrer"
        style="margin-left: 10px; margin-right: 10px; width: 100px; flex: 0 0 auto; align-self: flex-start;"
        target="_blank"
      >
        <img
          alt="Microsoft"
          data-nimg="1"
          decoding="async"
          height="50"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Flogos%2Fmicrosoft-logo.png&w=256&q=75"
          srcset="/_next/image?url=%2Fimages%2Flogos%2Fmicrosoft-logo.png&w=128&q=75 1x, /_next/image?url=%2Fimages%2Flogos%2Fmicrosoft-logo.png&w=256&q=75 2x"
          style="color: transparent;"
          width="100"
        />
      </a>
      <a
        href="http://www.google.com"
        rel="noreferrer"
        style="margin-left: 10px; margin-right: 10px; width: 100px; flex: 0 0 auto; align-self: flex-start;"
        target="_blank"
      >
        <img
          alt="LinkedIn"
          data-nimg="1"
          decoding="async"
          height="50"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Flogos%2FlinkedIn-logo.png&w=256&q=75"
          srcset="/_next/image?url=%2Fimages%2Flogos%2FlinkedIn-logo.png&w=128&q=75 1x, /_next/image?url=%2Fimages%2Flogos%2FlinkedIn-logo.png&w=256&q=75 2x"
          style="color: transparent;"
          width="100"
        />
      </a>
      <a
        href="http://www.google.com"
        rel="noreferrer"
        style="margin-left: 10px; margin-right: 10px; width: 100px; flex: 0 0 auto; align-self: flex-start;"
        target="_blank"
      >
        <img
          alt="Google"
          data-nimg="1"
          decoding="async"
          height="50"
          loading="lazy"
          src="/_next/image?url=%2Fimages%2Flogos%2Fgoogle-logo.png&w=256&q=75"
          srcset="/_next/image?url=%2Fimages%2Flogos%2Fgoogle-logo.png&w=128&q=75 1x, /_next/image?url=%2Fimages%2Flogos%2Fgoogle-logo.png&w=256&q=75 2x"
          style="color: transparent;"
          width="100"
        />
      </a>
    </div>
  </div>
</div>
`;
